.github/workflows/run-tests.yml

```diff
@@ -20,7 +20,7 @@ jobs:
         name: Job Two
         runs-on: ubuntu-latest
         needs: job-one 
-        #if: ${{ github.event_name == 'pull_request' && github.ref == 'refs/heads/main'}}
+        if: ${{ github.event_name == 'pull_request' && github.ref == 'refs/heads/main'}}
```

Comment:
`pull_request` event will trigger on PR event being created.
`push`, which happens when the code is merged in by completing the PR, may be what you want here